test/fs/test_fs_js_api.c

```diff
@@ -37,6 +37,47 @@ int main() {
 #endif
     );
 
+    /********** test FS.read() **********/
+    EM_ASM(
+        FS.writeFile("readtestfile", 'a=1_b=2_');
+
+        var buf = new Uint8Array(8);
+        var stream = FS.open("readtestfile", "r");
+        var numRead = FS.read(stream, buf, 0, 4);
+        assert(numRead == 4);
+        assert((new TextDecoder().decode(buf.subarray(0, 4))) === "a=1_");
+        numRead = FS.read(stream, buf, 4, 4);
+        assert(numRead == 4);
+        assert((new TextDecoder().decode(buf)) == 'a=1_b=2_');
+
+        stream = FS.open("readtestfile", "r");
```

Comment:
Maybe add a comment for the subsections. Like here, it took me a moment to realize the point of the line 55 was that it uses pread since it has the 5th param.

Reply:
(and that therefore the read after it begins at offset 0, since line 55 didn't affect the file position)